### PR TITLE
multus support

### LINKED ISSUE #789: multus support
### Your current KubeKey version

_No response_

### Describe this feature

Multus CNI is a container network interface (CNI) plugin for Kubernetes that enables attaching multiple network interfaces to pods. Typically, in Kubernetes each pod only has one network interface (apart from a loopback) -- with Multus you can create a multi-homed pod that has multiple interfaces. This is accomplished by Multus acting as a "meta-plugin", a CNI plugin that can call multiple other CNI plugins.

### Describe the solution you'd like

kk can deploy.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by chaunceyjiang**
multus support
FIx #789

### CODE REVIEW

**Comment by 24sama** [comment]
Thanks for your contribution! I will spend some time checking this PR.

File: pkg/config/templates/cluster.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ spec:
     version: {{ .Options.KubeVersion }}
     clusterName: cluster.local
   network:
+    # multus support. https://github.com/k8snetworkplumbingwg/multus-cni
+    # for example: plugin: calico-multus , plugin: flannel-multus , plugin: cilium-multus
```
Comment (pixiake) [inline pkg/config/templates/cluster.go:57]: I think it is better to use a parameter alone to enable `multus-cni`. such as: `enableMultusCNI: true`
If the configuration is mixed with other network plugins, the subsequent addition of new network plugins will also involve `multus-cni`.

Comment (chaunceyjiang) [reply pkg/config/templates/cluster.go:57]: You are right. I can't agree with you more

Comment (chaunceyjiang) [reply pkg/config/templates/cluster.go:57]: Does the `kk`  cli  need to add this parameter?

Comment (pixiake) [reply pkg/config/templates/cluster.go:57]: I think the current configuration file supports it. After all, cli does not currently support specifying calico, flannel, etc.

Comment (chaunceyjiang) [reply pkg/config/templates/cluster.go]: ok

File: pkg/common/common.go [hunk 2]
```diff
@@ -54,6 +54,11 @@ const (
 	Cilium  = "cilium"
 	Kubeovn = "kubeovn"
 
+	Calico_Multus  = "calico-multus"
+	Flannel_Multus = "flannel-multus"
+	Cilium_Multus  = "cilium-multus"
+	Kubeovn_Multus = "kubeovn-multus"
```
Comment (pixiake) [inline pkg/common/common.go:60]: I think these consts can also be deleted.

File: pkg/images/tasks.go [hunk 3]
```diff
@@ -85,7 +85,12 @@ func GetImage(runtime connector.ModuleRuntime, kubeConf *common.KubeConf, name s
 	} else {
 		corednsTag = "1.8.0"
 	}
-
+	multusRepoAddr := kubeConf.Cluster.Registry.PrivateRegistry
+	if multusRepoAddr == "" {
+		// TODO
```
Comment (pixiake) [inline pkg/images/tasks.go:90]: The image of multus has been synchronized to repo of kubesphere. 
Maybe `kubesphere/multus-cni:v3.8` can be used directly here.

File: pkg/common/common.go (hunk 2 above)
Comment (chaunceyjiang) [reply pkg/common/common.go]: sorry

File: pkg/plugins/network/modules.go [hunk 4]
```diff
@@ -50,6 +50,42 @@ func (d *DeployNetworkPluginModule) Init() {
...
+}
+
+func deployMultus(d *DeployNetworkPluginModule) []task.Interface {
+	generateMultus := &task.RemoteTask{
+		Name:  "GenerateMultus",
+		Desc:  "Generate multus cni",
+		Hosts: d.Runtime.GetHostsByRole(common.Master),
+		Prepare: &prepare.PrepareCollection{
+			new(common.OnlyFirstMaster),
+			&OldK8sVersion{Not: true},
```
Comment (24sama) [inline pkg/plugins/network/modules.go:65]: If the `multus-cni` support below k8s v1.16 version, maybe this `OldK8sVersion prepare` can be deleted.

File: pkg/images/tasks.go [hunk 5]
```diff
@@ -105,6 +104,7 @@ func GetImage(runtime connector.ModuleRuntime, kubeConf *common.KubeConf, name s
 		"cilium":                  {RepoAddr: kubeConf.Cluster.Registry.PrivateRegistry, Namespace: "cilium", Repo: "cilium", Tag: kubekeyv1alpha2.DefaultCiliumVersion, Group: kubekeyv1alpha2.K8s, Enable: strings.EqualFold(kubeConf.Cluster.Network.Plugin, "cilium")},
 		"operator-generic":        {RepoAddr: kubeConf.Cluster.Registry.PrivateRegistry, Namespace: "cilium", Repo: "operator-generic", Tag: kubekeyv1alpha2.DefaultCiliumVersion, Group: kubekeyv1alpha2.K8s, Enable: strings.EqualFold(kubeConf.Cluster.Network.Plugin, "cilium")},
 		"kubeovn":                 {RepoAddr: kubeConf.Cluster.Registry.PrivateRegistry, Namespace: "kubeovn", Repo: "kube-ovn", Tag: kubekeyv1alpha2.DefaultKubeovnVersion, Group: kubekeyv1alpha2.K8s, Enable: strings.EqualFold(kubeConf.Cluster.Network.Plugin, "kubeovn")},
+		"multus":                  {RepoAddr: kubeConf.Cluster.Registry.PrivateRegistry, Namespace: "k8snetworkplumbingwg", Repo: "multus-cni", Tag: kubekeyv1alpha2.DefalutMultusVersion, Group: kubekeyv1alpha2.K8s, Enable: strings.Contains(kubeConf.Cluster.Network.Plugin, "multus")},
```
Comment (pixiake) [inline pkg/images/tasks.go:107]: I think the value of  `Namespace` here can be replaced by `kubekeyv1alpha2.DefaultKubeImageNamespace`.

File: pkg/plugins/network/modules.go (hunk 4 above)
Comment (chaunceyjiang) [reply pkg/plugins/network/modules.go:65]: https://github.com/kubesphere/kubekey/blob/master/README.md#kubernetes-versions

kubekey not support below k8s v1.17.9 version,

File: pkg/images/tasks.go (hunk 5 above)
Comment (chaunceyjiang) [reply pkg/images/tasks.go:107]: you are right.

File: pkg/plugins/network/modules.go (hunk 4 above)
Comment (24sama) [reply pkg/plugins/network/modules.go:65]: In fact, kk support installing k8s v1.15.12. But naturally, we do not suggest installing this k8s version.

https://github.com/kubesphere/kubekey/blob/master/docs/kubernetes-versions.md

Comment (chaunceyjiang) [reply pkg/plugins/network/modules.go:65]: `multus-cni:v3.8`  CRD use `apiextensions.k8s.io/v1`  ,k8s v1.15.12  not support.

Comment (24sama) [reply pkg/plugins/network/modules.go:65]: Thanks for your research. It explains my confusion.

**Comment by 24sama** [comment]
Thanks for your contributions again. And there's the last thing, please squash your commits into one commit.

**Comment by 24sama** [comment]
@chaunceyjiang 
I think you don't need to close this PR, you just need to use commands `git rebase -i head~3` and modify the config, then use command `git push --force`.

**Pull request closed without merging**